docs/syntax-attributes-functions-parameters.md

```diff
@@ -0,0 +1,61 @@
+
+1. Inside the attribute list, `@[]`, attribute names are not allowed to have a leading `@`.
+2. Attribute lists cannot be nested. Attributes cannot be specified on attribute parameters.
+3. Attributes inside `@[]` are separated by a `,`.
+4. Attributes inside `@[]` are allowed to take an optional table of parameters.
+5. The "parameter table" is a Luau table whose entries specify attribute parameters, with or without names, separated by commas or semicolons.
+6. A parameter can be any literal value: `BOOLEAN`, `NUMBER`, `STRING`, `NIL`,  or a `parameter-table`. Attributes can be seen as a way of providing tagged metadata with no evaluation semantics of their own. With that in mind, we only allow these syntactic categories because they evaluate to themselves. This is also the reason we disallow entries of the form `[exp1] = [exp2]` in parameter table, since this would require evaluating `exp1` at construction time. `nil` is allowed for the sake of generality and completeness.
+7. Attributes can be specified before function declarations in multiple ways: `@attr1 @[attr2, attr3{2, "hi"}]`, `@attr1 @attr2 @[attr3{2, "hi"}]`, `@attr1 @[attr3{2, "hi"}] @attr2`, and  `@[attr1, attr2, attr3{2, "hi"}]` are all equivalent.
+8. An attribute with an empty parameter list can be equivalently specified without one, i.e., `@[attr{}]` is same as `@attr`.
+
+The parser is responsible for for enforcing the syntax specified by this RFC and ensuring that attributes are not repeated on a definition or a declaration. The number and type of parameters accepted by a particular attribute will be specified by its own RFC, and enforced by the relevant component of the implementation (typechecker, compiler, etc.), after parsing.
+
+We currently have an ad-hoc implementation of `@checked` attribute, used in declaration files. It is specified in two ways:
+
+1. `declare function @checked abs(n: number): number`
+2. `writef64: @checked (b: buffer, offset: number, value: number) -> ()`
+
+To ensure uniformity with the specification of this RFC, declarations using the first style will be modified to use `@checked` attribute before the `function` keyword. Declarations using the second style will remain unchanged.
+
+An alternative parameter syntax would be to implement function-style attribute syntax of the form `@attr(parameters, ...)`, but that would not allow specifying parameter names. Attribute syntax of the form `@attr {...}` will create ambiguity when attributes without parameters are used on tables, should we decide to allow attributes on them in the future. An advantage of the `@[]` syntax is that it provides the `]` delimiter which marks the end of attribute, leaving open the possibility of syntax evolution without introducing parsing ambiguity.
```

Comment:
Looks like this belongs in `Alternatives` section.

Reply:
Its not really an "Alternative" in the sense that its not an alternative to the attribute feature, but rather a discussion of why an obvious syntax was not adopted. The presentation has been changed to avoid this confusion.